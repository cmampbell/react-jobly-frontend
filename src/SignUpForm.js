import React, { useState } from 'react'
import { Form, useOutletContext } from 'react-router-dom'

const SignUpForm = () => {
    const { register } = useOutletContext()
    const INITIAL_STATE = {
        username: '',
        password: '',
        firstName: '',
        lastName: '',
        email: ''
    };
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (evt) => {
        const { name, value } = evt.target
        setFormData(() => ({
            ...formData, [name]: value }
        ))
    }

    const handleSubmit = (evt) => {
        evt.preventDefault();
        register(formData);
    }

    return (
        <Form onSubmit={handleSubmit}>
            <label htmlFor='username'>Username:</label>
            <input 
                placeholder='username'
                name="username" 
                id="username" 
                type="text" 
                value={formData.username}
                onChange={handleChange}/>
            <label htmlFor='password'>Password:</label>
            <input 
                placeholder='password'
                name="password"
                id="password"
                type="password"
                value={formData.password}
                onChange={handleChange}/>
            <label htmlFor='firstName'>First name:</label>
            <input 
                placeholder='first name'
                name="firstName" 
                id="firstName" 
                type="text" 
                value={formData.firstName}
                onChange={handleChange}/>
            <label htmlFor='lastName'>Last name:</label>
            <input 
                placeholder='last name'
                name="lastName" 
                id="lastName" 
                type="text" 
                value={formData.lastName}
                onChange={handleChange}/>
            <label htmlFor='email'>Email:</label>
            <input 
                placeholder='email'
                name="email" 
                id="email" 
                type="text" 
                value={formData.email}
                onChange={handleChange}/>
            <button type="submit">Sign Up!</button>
        </Form>
    )
}

export default SignUpForm;