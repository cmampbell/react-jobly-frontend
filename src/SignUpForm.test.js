import { render } from '@testing-library/react';
import SignUpForm from './SignUpForm';
import { RouterProvider, createMemoryRouter} from 'react-router-dom';

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useOutletContext: function () {
        return {
            register: () => console.log('registered')
        }
    }
})
);

const route = [
    {
      path: "/",
      element: <SignUpForm />,
    },
  ];

const router = createMemoryRouter(route);

describe('Smoke and snapshot CompanyCard test', () => {
    it('should render without crashing', () => {
        render(<RouterProvider router={router}/>);
    });
    it('should match snapshot', async ()=> {
        const { asFragment, findByText } = render(<RouterProvider router={router}/>);
        expect(asFragment()).toMatchSnapshot();
        expect(await findByText('Sign Up!')).toBeInTheDocument()
    })
})