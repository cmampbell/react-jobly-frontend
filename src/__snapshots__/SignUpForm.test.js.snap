// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke and snapshot CompanyCard test should match snapshot 1`] = `
<DocumentFragment>
  <form
    action="/"
    method="get"
  >
    <label
      for="username"
    >
      Username:
    </label>
    <input
      id="username"
      name="username"
      placeholder="username"
      type="text"
      value=""
    />
    <label
      for="password"
    >
      Password:
    </label>
    <input
      id="password"
      name="password"
      placeholder="password"
      type="password"
      value=""
    />
    <label
      for="firstName"
    >
      First name:
    </label>
    <input
      id="firstName"
      name="firstName"
      placeholder="first name"
      type="text"
      value=""
    />
    <label
      for="lastName"
    >
      Last name:
    </label>
    <input
      id="lastName"
      name="lastName"
      placeholder="last name"
      type="text"
      value=""
    />
    <label
      for="email"
    >
      Email:
    </label>
    <input
      id="email"
      name="email"
      placeholder="email"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Sign Up!
    </button>
  </form>
</DocumentFragment>
`;
